modules/auxiliary/gather/behind_cloudflare.rb

```diff
@@ -0,0 +1,445 @@
+      ])
+  end
+
+  def do_check_tcp_port(ip, port, proxies)
+    begin
+      sock = Rex::Socket::Tcp.create(
+        'PeerHost' => ip,
+        'PeerPort' => port,
+        'Proxies'  => proxies
+      )
+    rescue ::Rex::ConnectionRefused, Rex::ConnectionError
+      return false
+    end
+    sock.close
+    return true
+  end
+
+  def do_grab_domain_ip_history(hostname, proxies)
+    begin
+      cli = Rex::Proto::Http::Client.new('viewdns.info', 443, {}, true, nil, proxies)
+      cli.connect
+
+      request = cli.request_cgi({
+        'uri'    => "/iphistory/?domain=#{hostname}",
+        'method' => 'GET'
+      })
+      response = cli.send_recv(request)
+      cli.close
+
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT
+      print_error('HTTP connection failed to ViewDNS.info website.')
+      return false
+    end
+
+    html  = response.get_html_document
+    table = html.css('table')[2]
+
+    unless table.nil?
+      rows   = table.css('tr')
+
+      ar_ips = []
+      rows.each.map do | row |
+        row = /(\d*\.\d*\.\d*\.\d*)/.match(row.css('td').map(&:text).to_s)
+        unless row.nil?
+          ar_ips.push(row)
+        end
+      end
+    end
+
+    if ar_ips.nil?
+      print_bad('No domain IP(s) history founds.')
+      return false
+    end
+
+    return ar_ips
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_enumeration(domain, threads)
+    wordlist = datastore['WORDLIST']
+    return if wordlist.blank?
+    threads  = 1 if threads <= 0
+
+    queue    = []
+    File.foreach(wordlist) do | line |
+      queue << "#{line.chomp}.#{domain}"
+    end
+
+    ar_ips = []
+    until queue.empty?
+      t = []
+      threads = 1 if threads <= 0
+
+      if queue.length < threads
+        # work around issue where threads not created as the queue isn't large enough
+        threads = queue.length
+      end
+
+      begin
+        1.upto(threads) do
+          t << framework.threads.spawn("Module(#{refname})", false, queue.shift) do | test_current |
+            Thread.current.kill unless test_current
+            a = /(\d*\.\d*\.\d*\.\d*)/.match(do_dns_get_a(test_current, 'DNS bruteforce records').to_s)
+            ar_ips.push(a) if a
+          end
+        end
+        t.map(&:join)
+
+      rescue ::Timeout::Error
+      ensure
+        t.each { | x | x.kill rescue nil }
+      end
+    end
+
+    if ar_ips.empty?
+      print_bad('No enumerated domain IP(s) founds.')
+      return false
+    end
+
+    return ar_ips
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_get_a(domain, type='DNS A records')
+    response = do_dns_query(domain, 'A')
+    return if response.blank? || response.answer.blank?
+
+    response.answer.each do | row |
+      next unless row.class == Net::DNS::RR::A
+    end
+  end
+
+  ## auxiliary/gather/enum_dns.rb
+  def do_dns_query(domain, type)
+    begin
```

Comment:
Rather than wrap the entire method in a `begin` / `rescue` block, you can use the rescue-from-method technique.

```ruby
  ## auxiliary/gather/enum_dns.rb
  def do_dns_query(domain, type)
    nameserver = datastore['NS']

    if nameserver.blank?
      dns = Net::DNS::Resolver.new
    else
      dns = Net::DNS::Resolver.new(nameservers: ::Rex::Socket.resolv_to_dotted(nameserver))
    end

    dns.use_tcp        = false
    dns.udp_timeout    = 8
    dns.retry_number   = 2
    dns.retry_interval = 2
    dns.query(domain, type)
  rescue ResolverArgumentError, Errno::ETIMEDOUT, ::NoResponseError, ::Timeout::Error => e
    print_error("Query #{domain} DNS #{type} - exception: #{e}")
    return nil
  end
```